### PR TITLE
Updates to CrvReco and CrvCoincidenceFinder

### PR DESCRIPTION
**Pull request by ehrlich-uva**
-added new and removed unused user options at the CrvReco module
-added status flags to the CrvReco pulse
-updated CrvCoincidenceFinder w.r.t. the above changes
-added option to CrvCoincidenceFinder to simplify the process for very large clusters

### CODE REVIEW

**Review by brownd1978 — APPROVED** [review]
This looks good Ralf.

File: CRVResponse/fcl/prolog_v10.fcl [hunk 1]
```diff
@@ -146,10 +147,12 @@ BEGIN_PROLOG
                                         //e.g. 0.985 for a leading edge at 50% pulse height
                                         //e.g. 1.385 for a leading edge at 20% pulse height
                                         //e.g. 1.587 for a leading edge at 10% pulse height
-      allowDoubleGumbel         : false //tries fitting with two Gumbel functions (needs a very small "minPulseHeightRatio", e.g. 0.01)
-      doubleGumbelThreshold     : 2.0   //Chi2/#ADCsamples (based on a single Gumble fit)
-                                        //at which a fit with two Gumbel functions should be attempted
-      minPEs                    : 0     //0 PEs
+      pulseThreshold            : 0.5   //50% of ADC peak value above pedestal
+                                        //threshold used to determine the pulse time interval for the no-fit option
+      pulseAreaThreshold        : 5     //5 ADC above pedestal
```
Comment (oksuzian) [inline CRVResponse/fcl/prolog_v10.fcl:152]: Is it really an area threshold as the name suggests?
We can't zero suppress the data on the charge.

File: CRVResponse/fcl/prolog_v10.fcl [hunk 2]
```diff
@@ -146,10 +147,12 @@ BEGIN_PROLOG
                                         //e.g. 0.985 for a leading edge at 50% pulse height
                                         //e.g. 1.385 for a leading edge at 20% pulse height
                                         //e.g. 1.587 for a leading edge at 10% pulse height
-      allowDoubleGumbel         : false //tries fitting with two Gumbel functions (needs a very small "minPulseHeightRatio", e.g. 0.01)
-      doubleGumbelThreshold     : 2.0   //Chi2/#ADCsamples (based on a single Gumble fit)
-                                        //at which a fit with two Gumbel functions should be attempted
-      minPEs                    : 0     //0 PEs
+      pulseThreshold            : 0.5   //50% of ADC peak value above pedestal
```
Comment (oksuzian) [inline CRVResponse/fcl/prolog_v10.fcl:150]: Please add more info on what these variable are

File: CRVResponse/test/CRVResponseReco.fcl [hunk 3]
```diff
@@ -15,11 +15,12 @@ source :
...
-  RandomNumberGenerator: {defaultEngineKind: "MixMaxRng" }
-  GeometryService        : { inputFile : "Offline/Mu2eG4/geom/geom_common.txt" }
-  ConditionsService      : { conditionsfile : "Offline/ConditionsService/data/conditions_01.txt" }
-  GlobalConstantsService : { inputFile : "Offline/GlobalConstantsService/data/globalConstants_01.txt" }
-  SeedService            : @local::automaticSeeds
+#  RandomNumberGenerator: {defaultEngineKind: "MixMaxRng" }
+#  GeometryService        : { inputFile : "Offline/Mu2eG4/geom/geom_common.txt" }
+#  ConditionsService      : { conditionsfile : "Offline/ConditionsService/data/conditions_01.txt" }
+#  GlobalConstantsService : { inputFile : "Offline/GlobalConstantsService/data/globalConstants_01.txt" }
+#  SeedService            : @local::automaticSeeds
```
Comment (oksuzian) [inline CRVResponse/test/CRVResponseReco.fcl:22]: Should we remove obsolete lines?

**Review by oksuzian — APPROVED** [review]
Looks good. The algorithm needs to be checked, which we'll do...

File: CRVResponse/fcl/prolog_v10.fcl (hunk 1 above)
Comment (ehrlich-uva) [reply CRVResponse/fcl/prolog_v10.fcl:152]: This threshold determines the range where the ADC values of a pulse are summed up. If the waveform crosses this threshold at the beginning of the pulse, we start adding the ADC counts to calculate the PE value (in the no-fit option) until the waveform falls below this threshold at the end of the pulse. I should change the explanation in the fcl file.

File: CRVResponse/fcl/prolog_v10.fcl (hunk 2 above)
Comment (ehrlich-uva) [reply CRVResponse/fcl/prolog_v10.fcl:150]: Will do.

File: CRVResponse/test/CRVResponseReco.fcl (hunk 3 above)
Comment (ehrlich-uva) [reply CRVResponse/test/CRVResponseReco.fcl:22]: Yes, I will remove these lines.

File: CRVResponse/fcl/prolog_v10.fcl (hunk 1 above)
Comment (brownd1978) [reply CRVResponse/fcl/prolog_v10.fcl:152]: Rob, are you suggesting he remove these comments?  I personally find a description of the meaning of the variable helpful.  Maybe you mean he should remove the value from the comment?

Comment (kutschke) [reply CRVResponse/fcl/prolog_v10.fcl:152]: Perhaps I misundertood but I understood the request to be for something that, if done properly, would be many lines long.  If a meaningful short comment will do the job, then I agree that having it inline is preferred.  I was imagining a case in which a meaningful comment would be many lines long.

Comment (kutschke) [reply CRVResponse/fcl/prolog_v10.fcl:152]: This is just a suggestion, not a requirement.  So I will approve.

**Review by kutschke — APPROVED** [review]
I did not do a thorough review - I only followed up on one of the discussions.

**Comment by brownd1978** [comment]
@FNALbuild run build test

**Pull request merged**